Gas mixtures in porous flow

cpgr opened the pull request:
I've cleaned up @ChiCoTran's original PR that had been closed due to my inaction.

Refs #26795

ChiCoTran committed:
Add material for arbitrary gas mixutres in porous flow

Refs #26795

cpgr committed:
Make tests more robust and update requirements

Refs #26795

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2458468/) on b85f19c wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/28682/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2458515/) on b85f19c wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Porous flow

<table>
  <thead>
    <tr>
      <th></th>
      <th>1da09a</th>
      <th colspan=4>#28682 b85f19</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>95.30%</td>
      <td>95.40%</td>
      <td>+0.10%</td>
      <td>96.19%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>11336</td>
      <td>11560</td>
      <td>+224</td>
      <td>303</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>559</td>
      <td>557</td>
      <td>-2</td>
      <td>12</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/28682/coverage_diff/porous_flow)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/28682/coverage/porous_flow)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/xfem)

</details>



This comment will be updated on new commits.

cpgr committed:
Refactor code for brevity and correctness

Included additional chain rule derivative and clean up logic to
reduce duplication

Refs #26795

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2544235/), step Docs: sync website on 5e63615 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/28682/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2544282/), step Generate coverage on 5e63615 wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>d3e2af</th>
      <th colspan=4>#28682 5e6361</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.09%</td>
      <td>84.94%</td>
      <td>-0.15%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>106774</td>
      <td>106585</td>
      <td>-189</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>18703</td>
      <td>18892</td>
      <td>+189</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/28682/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/28682/coverage/framework)


## Modules coverage

### Porous flow

<table>
  <thead>
    <tr>
      <th></th>
      <th>d3e2af</th>
      <th colspan=4>#28682 5e6361</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>95.30%</td>
      <td>95.16%</td>
      <td>-0.14%</td>
      <td>86.11%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>11336</td>
      <td>11505</td>
      <td>+169</td>
      <td>248</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>559</td>
      <td>585</td>
      <td>+26</td>
      <td>40</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/28682/coverage_diff/porous_flow)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/28682/coverage/porous_flow)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/stochastic_tools)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/28682/coverage/xfem)

</details>

## Warnings

- `porous_flow` new line coverage rate 86.11% is less than the suggested 90.0%



This comment will be updated on new commits.

cpgr committed:
Refactor gas mixture to use AD for future derived classes

Update tests, examples and documentation.

Closes #26795

cpgr committed:
Remove superfluous template in paramError

Refs #26795
